apex/transformer/pipeline_parallel/schedules/fwd_bwd_pipelining_without_interleaving.py

```diff
@@ -456,6 +475,9 @@ def forward_backward_pipelining_without_interleaving(
     _logger.info("Cooldown phase")
     if not forward_only:
         for i in range(num_warmup_microbatches):
+            if i == num_warmup_microbatches-1 and rank == 0:
+                context.__exit__(None, None, None)
+                context = None
```

Comment:
Question: why do we need to set `context` to `None`?

Reply:
I want to check at the end if the context has been exited, but I could also do it by setting a bool. Do you think that's easier to understand?

Reply:
hmm, I cannot describe why but this value assignment doesn't feel nice